File: trunk/AI/RAI/python3-cgdk/Runner.py
import sys
from MyStrategy import MyStrategy
from RemoteProcessClient import RemoteProcessClient
from model.Move import Move

class Runner:
    def __init__(self):
        if sys.argv.__len__() == 4:
            self.remote_process_client = RemoteProcessClient(sys.argv[1], int(sys.argv[2]))
            self.token = sys.argv[3]
        else:
            self.remote_process_client = RemoteProcessClient("localhost", 31000)
            self.token = "0000000000000000"

    def run(self):
        try:
            self.remote_process_client.write_token(self.token)
            team_size = self.remote_process_client.read_team_size()

            strategies = []
            tank_types = []

            for strategy_index in range(team_size):
                strategy = MyStrategy()
                strategies.append(strategy)
                tank_types.append(strategy.select_tank(strategy_index, team_size))

            self.remote_process_client.write_selected_tanks(tank_types)

            while True:
                player_context = self.remote_process_client.read_player_context()
                if player_context is None:
                    break

                player_tanks = player_context.tanks
                if player_tanks.__len__() != team_size:
                    break

                moves = []

                for strategy_index in range(team_size):
                    move = Move()
                    moves.append(move)
                    strategies[strategy_index].move(player_tanks[strategy_index], player_context.world, move)

                self.remote_process_client.write_moves(moves)
        finally:
            self.remote_process_client.close()

Runner().run()

File: AI/RAI/cpp-cgdk/Runner.cpp
#include "Runner.h"

#include <vector>

#include "src/MyStrategy.h"
#include "Strategy.h"

using namespace model;
using namespace std;

int main(int argc, char* argv[]) {
	if (argc == 4) {
		Runner runner(argv[1], argv[2], argv[3]);
		runner.Run();
	} else {
		Runner runner("localhost", "31000", "0000000000000000");
		runner.Run();
	}
	
	return 0;
}

Runner::Runner(const char* host, const char* port, const char* token) : remote_process_client(host, atoi(port)), token(token) {
}

void Runner::Run() {
	remote_process_client.WriteToken(token);
    int team_size = remote_process_client.ReadTeamSize();

    vector<Strategy*> strategies;
    vector<TankType> tank_types;

	for (int strategy_index = 0; strategy_index < team_size; ++strategy_index) {
        Strategy* strategy = new MyStrategy;
		strategies.push_back(strategy);
		tank_types.push_back(strategy->SelectTank(strategy_index, team_size));
    }

    remote_process_client.WriteSelectedTanks(tank_types);

    PlayerContext* player_context;

    while ((player_context = remote_process_client.ReadPlayerContext()) != NULL) {
        vector<Tank> player_tanks = player_context->tanks();
        if (player_tanks.size() != team_size) {
            break;
        }

        vector<Move> moves;

        for (int strategy_index = 0; strategy_index < team_size; ++strategy_index) {
            Move move;
            strategies[strategy_index]->Move(player_tanks[strategy_index], player_context->world(), move);
			moves.push_back(move);
        }

        remote_process_client.WriteMoves(moves);

		delete player_context;
    }

	for (int strategy_index = 0; strategy_index < team_size; ++strategy_index) {
		delete strategies[strategy_index];
	}
}
